l1-contracts/test/NewOutbox.t.sol

```diff
@@ -0,0 +1,225 @@
+    outbox.consume(1, 0, fakeMessage, path);
+  }
+
+  function testRevertIfPathHeightMismatch() public {
+    DataStructures.L2ToL1Msg memory fakeMessage = _fakeMessage(address(this));
+    bytes32 leaf = fakeMessage.sha256ToField();
+
+    NaiveMerkle tree = new NaiveMerkle(_DEFAULT_TREE_HEIGHT);
+    tree.insertLeaf(leaf);
+    bytes32 root = tree.computeRoot();
+
+    vm.prank(_STATE_TRANSITIONER);
+    outbox.insert(1, root, _DEFAULT_TREE_HEIGHT);
+
+    NaiveMerkle biggerTree = new NaiveMerkle(_DEFAULT_TREE_HEIGHT + 1);
+    tree.insertLeaf(leaf);
+
+    (bytes32[] memory path,) = biggerTree.computeSiblingPath(0);
+    vm.expectRevert(
+      abi.encodeWithSelector(
+        Errors.Outbox__InvalidPathLength.selector, _DEFAULT_TREE_HEIGHT, _DEFAULT_TREE_HEIGHT + 1
+      )
+    );
+    outbox.consume(1, 0, fakeMessage, path);
+  }
+
+  function testRevertIfTryingToConsumeMessageNotInTree() public {
+    DataStructures.L2ToL1Msg memory fakeMessage = _fakeMessage(address(this));
+    bytes32 leaf = fakeMessage.sha256ToField();
+    fakeMessage.content = bytes32(uint256(42069));
+    bytes32 modifiedLeaf = fakeMessage.sha256ToField();
+
+    NaiveMerkle tree = new NaiveMerkle(_DEFAULT_TREE_HEIGHT);
+    tree.insertLeaf(leaf);
+    bytes32 root = tree.computeRoot();
+
+    NaiveMerkle modifiedTree = new NaiveMerkle(_DEFAULT_TREE_HEIGHT);
+    modifiedTree.insertLeaf(modifiedLeaf);
+    bytes32 modifiedRoot = modifiedTree.computeRoot();
+
+    vm.prank(_STATE_TRANSITIONER);
+    outbox.insert(1, root, _DEFAULT_TREE_HEIGHT);
+
+    (bytes32[] memory path,) = modifiedTree.computeSiblingPath(0);
+
+    vm.expectRevert(abi.encodeWithSelector(Errors.Outbox__InvalidRoot.selector, root, modifiedRoot));
+    outbox.consume(1, 0, fakeMessage, path);
+  }
+
+  function testInsertVariedLeafs(bytes32[] calldata _messageLeafs) public {
+    vm.assume(_messageLeafs.length < 256);
+    vm.assume(_messageLeafs.length > 128);
+
+    // This correct only when fuzzing using _messageLeafs.length 128 < length < 256
+    NaiveMerkle tree = new NaiveMerkle(8);
+
+    for (uint256 i = 0; i < _messageLeafs.length; i++) {
+      vm.assume(_messageLeafs[i] != bytes32(0));
+      tree.insertLeaf(_messageLeafs[i]);
+    }
+
+    bytes32 root = tree.computeRoot();
+
+    vm.expectEmit(true, true, true, true, address(outbox));
+    emit RootAdded(1, root, 8);
+    vm.prank(_STATE_TRANSITIONER);
+    outbox.insert(1, root, 8);
+  }
+
+  // This test takes awhile so to keep it somewhat reasonable wev'e set a limit on the amount of fuzz runs
+  /// forge-config: default.fuzz.runs = 16
+  function testInsertAndConsumeWithVariedRecipients(
+    address[] calldata _recipients,
+    uint256 _blockNumber
+  ) public {
+    // We most likely will not have > 128 L2 -> L1 Messages in a single block but are testing an upper bound here
+    vm.assume(_recipients.length < 256);
+    vm.assume(_recipients.length > 128);
```

Comment:
See my prior comment about `vm.assume`, particularly if you're having performance issues.